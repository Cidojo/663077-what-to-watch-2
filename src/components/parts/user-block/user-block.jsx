import * as React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';

import Selectors from './../../../selectors/selectors.js';
import {Url} from './../../../constants/constants.js';

const UserBlock = (props) => {
  const {userData} = props;

  const {avatarUrl, name} = userData;

  return (
    <div className="user-block">
      {Object.keys(userData).length > 0 ?
        <div className="user-block__avatar">
          <Link to={Url.MY_LIST}>
            <img src={`${/.*(?=\/wtw$)/.exec(Url.BASE)[0]}${avatarUrl}`} alt={name} width="63" height="63"/>
          </Link>
        </div> :
        <Link to={Url.LOGIN} className="user-block__link">Sign in</Link>
      }
    </div>
  );
};

UserBlock.propTypes = {
  userData: PropTypes.shape({
    avatarUrl: PropTypes.string,
    name: PropTypes.string
  })
};

UserBlock.defaultProps = {
  userData: {}
};

const mapStateToProps = (state, ownProps) => {
  return Object.assign({}, ownProps, {
    userData: Selectors.getUserData(state)
  });
};

export {UserBlock};
export default connect(mapStateToProps)(UserBlock);
