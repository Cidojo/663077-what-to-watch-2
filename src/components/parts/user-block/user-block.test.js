import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import {UserBlock} from './user-block.jsx';

const props = {
  userData: {}
};

it(`should render UserBlock component without errors`, () => {
  const tree = renderer.create(<BrowserRouter><UserBlock {...props} /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
