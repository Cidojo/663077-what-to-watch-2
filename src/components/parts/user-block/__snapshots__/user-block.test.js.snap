// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserBlock component without errors 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
